src/norm_flows.jl

```diff
@@ -0,0 +1,98 @@
+using Distributions
+using LinearAlgebra
+using Random
+using StatsFuns: softplus
+using Roots, LinearAlgebra # for inverse
+
+################################################################################
+#                            Planar and Radial Flows                           #
+#             Ref: Variational Inference with Normalizing Flows,               #
+#               D. Rezende, S. Mohamed(2015) arXiv:1505.05770                  #
+################################################################################
+
+mutable struct PlanarLayer <: Bijector
+    w
+    u
+    u_hat
+    b
+end
+
+mutable struct RadialLayer <: Bijector
+    α_
+    β
+    z_not
+end
+
+function update_u_hat(u, w)
+    # to preserve invertibility
+    u_hat = u + (planar_flow_m(transpose(w) * u) - transpose(w) * u)[1] * w/(norm(w[:,1],2) ^ 2)
+end
+
+function update_u_hat!(flow::PlanarLayer)
+    flow.u_hat = flow.u + (planar_flow_m(transpose(flow.w) * flow.u) - transpose(flow.w) * flow.u)[1] * flow.w/(norm(flow.w,2)^2)
+end
+
+
+function PlanarLayer(dims::Int)
+    w = param(randn(dims, 1))
+    u = param(randn(dims, 1))
+    b = param(randn(1))
+    u_hat = update_u_hat(u, w)
+    return PlanarLayer(w, u, u_hat, b)
+end
+
+function RadialLayer(dims::Int)
+    α_ = param(randn(1))
+    β = param(randn(1))
+    z_not = param(randn(dims, 1))
+    return RadialLayer(α_, β, z_not)
+end
+
+planar_flow_m(x) = -1 .+ log.(1 .+ exp.(x)) # For planar flow
+dtanh(x) = 1 .- (tanh.(x)).^2 # For planar flow
+ψ(z, w, b) = dtanh(transpose(w) * z .+ b) .* w # For planar flow
+h(α, r) = 1 ./ (α .+ r) # For radial flow
+dh(α, r) = -h(α, r).^2 # For radial flow
+
+function transform(flow::PlanarLayer, z)
+    return z + flow.u_hat*tanh.(transpose(flow.w)*z .+ flow.b)
+end
+
+function transform(flow::RadialLayer, z)
```

Comment:
It would be awesome if you could put an equation number in a comment here inside the transforms!:)

For example I was wondering why you implement the `inv` the way you do, and struggled to find it in the paper (it's in the appendix).